pkg/clients/aws/route53.go

```diff
@@ -123,13 +126,23 @@ func (c *awsClient) AnswerDNSChallenge(reqLogger logr.Logger, acmeChallengeToken
 		return "", err
 	}
 
-	baseDomain := cr.Spec.ACMEDNSDomain
-	if !strings.HasSuffix(baseDomain, ".") {
-		baseDomain = baseDomain + "."
+	dnsZones := hivev1.DNSZoneList{}
+	err = c.kubeClient.List(context.TODO(), &dnsZones, &client.ListOptions{Namespace: c.namespace})
+	if err != nil {
+		reqLogger.Error(err, err.Error())
+		return "", err
+	}
+
+	if len(dnsZones.Items) == 0 {
+		return "", fmt.Errorf("no dns zones found")
 	}
+	dnsZone := dnsZones.Items[0]
+
+	dnsZonePath := dnsZone.Status.AWS.ZoneID
+	dnsZoneId := filepath.Base(*dnsZonePath)
```

Comment:
```suggestion
	dnsZoneId := filepath.Base(*dnsZone.Status.AWS.ZoneID)
```

Does the following work to eliminate the need to create a new in-memory object for `dnsZonePath`?